src/main/kotlin/org/rust/lang/refactoring/extractFunction/RsExtractFunctionConfig.kt

```diff
@@ -150,12 +155,14 @@ class RsExtractFunctionConfig private constructor(
                 .filter { it.textOffset <= end }
 
             val parameters = letBindings
-                .filter { it.textOffset <= start }
-                .filter {
-                    ReferencesSearch.search(it, LocalSearchScope(fn))
-                        .any { ref -> ref.element.textOffset in start..end }
+                .mapNotNull {
+                    if (it.textOffset > start) return@mapNotNull null
+                    val result = ReferencesSearch.search(it, LocalSearchScope(fn))
+                    if (result.none { it.element.textOffset in start..end }) return@mapNotNull null
+                    val requiredBorrowing = result.any { it.element.textOffset > end }
```

Comment:
Let's use a little bit different strategy here - pass a variable with minimal required "attribute". I mean if it is enough to pass only reference, let's pass reference even we can move variable and don't break code.
So, for 
```rust
fn foo() {
    let mut v = Vec::new();
    <selection>v.push(123);</selection>
}
```
result should be
```rust
fn foo() {
    let mut v = Vec::new();
    bar(&mut v);
}

fn bar(v: &mut Vec<i32>) {
    v.push(123);
}
```

Reply:
Also current implementation can pass references by reference
```rust
fn foo() {
    let s = "str";
    <selection>println!("{}", s);</selection>
    println!("{}", s);
}
```
=>
```rust
fn foo() {
    let s = "str";
    bar(&s);
    println!("{}", s);
}

fn bar(s: &&str) {
    println!("{}", s);
}
```